static/script/devices/device.js

```diff
@@ -635,14 +635,58 @@ define(
              */
             loadAuthenticatedURL: function(/*url, opts*/) {
             },
+
+            /**
+             * Returns a new XMLHttpRequest. Overridden in unit tests.
+             * @returns {XMLHttpRequest} a new instance
+             * @private
+             */
+            _newXMLHttpRequest: function() {
```

Comment:
I started looking into using Jasmine AJAX for this, but we will need to upgrade to Jasmine 2.0 first - Jasmine AJAX for older versions only supports jQuery.

I will use sinon for this instead.